port/unix/omrsysinfo.c

```diff
@@ -5962,19 +6180,94 @@ isRunningInContainer(struct OMRPortLibrary *portLibrary, BOOLEAN *inContainer)
 	return rc;
 }
 
+/**
+ * Read the cgroup memory limit from the memory subsystem file. For cgroup v1, it
+ * will always be a positive integer. For cgroup v2, the value in the file is
+ * either a positive integer or "max" (which indicates no limit).
+ *
+ * @param[in] portLibrary pointer to OMRPortLibrary.
+ * @param[in] fileName name of the file under cgroup subsystem.
+ * @param[out] limit pointer to uint64_t which on successful return contains
+ *      the cgroup memory limit.
+ *
+ * @return 0 on success, otherwise negative error code.
+ */
+static int32_t
+readCgroupMemoryFileIntOrMax(struct OMRPortLibrary *portLibrary, const char *fileName, uint64_t *limit)
```

Comment:
@EricYangIBM Should we add error handling for `fileName` if it is NULL or invalid? Currently, `fileName` will flow all the way to `getAbsolutePathOfCgroupSubsystemFile`, and then `fopen` will fail. Catching and reporting the error early will have benefits for diagnosis and avoid an implicit failure. 

I plan to approve the PR after this feedback is addressed. After addressing this comment, can you also run downstream sanity testing + local cgroup v2 testing?

Reply:
How would we check that `fileName` is invalid? I think `fopen` failing is sufficient. I can add a check for null though

Reply:
`NULL` check for `fileName` should be good; similar to the `NULL` check for `cgroup`. `fopen` can be used for invalidity.

Reply:
Added to readCgroupSubsystemFile

Reply:
There are other locations where `getHandleOfCgroupSubsystemFile` can be invoked. The other locations are missed if the null check is added to `readCgroupSubsystemFile`. The null check should go inside `getAbsolutePathOfCgroupSubsystemFile` in order to cover all paths.